src/scripts/i18n/i18n_extraction.py

```
@@ -0,0 +1,358 @@
+    cleaned = resp_content.strip()
+    # Remove markdown code block markers (```json or ```)
+    if cleaned.startswith('```json'):
+        cleaned = cleaned[7:]
+    if cleaned.startswith('```'):
+        cleaned = cleaned[3:]
+    if cleaned.endswith('```'):
+        cleaned = cleaned[:-3]
+    cleaned = cleaned.strip('` \n')
+    return json.loads(cleaned)
+
+def process_batch(batch, batch_num, type):
```

Ruff (0.11.9)

Missing return type annotation for public function `process_batch`

(ANN201)